### PR TITLE
Feat/#30 setup room database

### PR DESCRIPTION
**Pull request by hoffmatteo**
Setup basic room database, RecyclerView in card design using DB, Detail Activity when card is pressed, also updated color

TODO: Dimens, Strings, etc. --> first version for tomorrow

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -3,16 +3,22 @@
...
+        <entry key="..\:/Users/matte/Desktop/OTH/Scrum/Plantico/app/src/main/res/layout/content_main.xml" value="0.35208333333333336" />
+        <entry key="..\:/Users/matte/Desktop/OTH/Scrum/Plantico/app/src/main/res/layout/fragment_first.xml" value="0.35208333333333336" />
+        <entry key="..\:/Users/matte/Desktop/OTH/Scrum/Plantico/app/src/main/res/layout/fragment_second.xml" value="0.35208333333333336" />
+        <entry key="..\:/Users/matte/Desktop/OTH/Scrum/Plantico/app/src/main/res/layout/plant_detailpage.xml" value="0.35208333333333336" />
+        <entry key="..\:/Users/matte/Desktop/OTH/Scrum/Plantico/app/src/main/res/layout/plant_item.xml" value="0.5" />
+        <entry key="..\:/Users/matte/Desktop/OTH/Scrum/Plantico/app/src/main/res/mipmap-anydpi-v26/image_cactures.xml" value="0.314" />
         <entry key="..\:/programmierte programme/githubworkspace/Plantico/app/src/main/res/layout/fragment_first.xml" value="0.14260385005065856" />
         <entry key="..\:/programmierte programme/githubworkspace/Plantico/app/src/main/res/layout/fragment_second.xml" value="0.14260385005065856" />
```
Comment (tobireith) [inline .idea/misc.xml]: Not sure if all of this stuff should even be here?

File: app/build.gradle [hunk 2]
```diff
@@ -48,4 +49,21 @@ dependencies {
...
+    implementation "androidx.lifecycle:lifecycle-viewmodel-ktx:2.4.1"
+    implementation "androidx.lifecycle:lifecycle-livedata-ktx:2.4.1"
+    implementation "androidx.lifecycle:lifecycle-common-java8:2.4.1"
+
+    // Kotlin components
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7"
+    api "org.jetbrains.kotlinx:kotlinx-coroutines-core"
+    api "org.jetbrains.kotlinx:kotlinx-coroutines-android"
+
+}
```
Comment (tobireith) [inline app/build.gradle]: No newline ;)

File: app/src/main/java/de/othr/plantico/MainActivity.kt [hunk 3]
```diff
@@ -2,10 +2,30 @@ package de.othr.plantico
...
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        testViewModel.allPlants.observe(this) { plants ->
+            // Update the cached copy of the words in the adapter.
+            plants.let { adapter.submitList(it) }
+        }
     }
+
+
+
 }
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/MainActivity.kt]: No newline ;)

File: app/src/main/java/de/othr/plantico/PlantActivity.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+
+        }
+    }
+
+    companion object {
+        const val SELECTED_PLANT = "selected_plant"
+
+    }
+}
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/PlantActivity.kt]: No newline ;)

File: app/src/main/java/de/othr/plantico/PlantListAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            return oldItem === newItem
+        }
+
+        override fun areContentsTheSame(oldItem: Plant, newItem: Plant): Boolean {
+            return oldItem.plantName == newItem.plantName
+        }
+    }
+
+
+}
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/PlantListAdapter.kt]: No newline ;)

File: app/src/main/java/de/othr/plantico/database/PlantApplication.kt [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import android.app.Application
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+
+class PlantApplication: Application() {
+    val applicationScope = CoroutineScope(SupervisorJob())
+
+    val database by lazy {PlanticoDB.getDatabase(this, applicationScope)}
+    val repository by lazy {Repository(database.plantDAO())}
+}
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/database/PlantApplication.kt]: No newline ;)

File: app/src/main/java/de/othr/plantico/database/PlanticoDB.kt [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                    .addCallback(PlantDatabaseCallback(scope))
+                    .build()
+                INSTANCE = instance
+                // return instance
+                instance
+
+            }
+        }
+    }
+}
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/database/PlanticoDB.kt]: No newline ;)

File: app/src/main/java/de/othr/plantico/database/Repository.kt [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+    @Suppress("RedundantSuspendModifier")
+    @WorkerThread
+    suspend fun insert(plant: Plant) {
+        plantDAO.insert(plant)
+    }
+
+
+
+}
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/database/Repository.kt]: No newline ;)

File: app/src/main/java/de/othr/plantico/database/daos/PlantDAO.kt [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+@Dao
+interface PlantDAO {
+    @Query("SELECT * FROM plant ORDER BY id")
+    fun getAllPlants(): Flow<List<Plant>>
+
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
+    suspend fun insert(plant: Plant)
+    @Query("DELETE FROM Plant")
+    suspend fun deleteAll()
+}
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/database/daos/PlantDAO.kt]: No newline ;)

File: app/src/main/java/de/othr/plantico/database/entities/Plant.kt [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Entity
+data class Plant(
+    @PrimaryKey(autoGenerate = true) val id: Int,
+    val plantName: String
+) {
+}
```
Comment (tobireith) [inline app/src/main/java/de/othr/plantico/database/entities/Plant.kt]: No newline ;)

File: app/src/main/res/layout/activity_plant.xml [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        android:text="spaceholder"
+        android:layout_gravity="center"
+        android:textAlignment="center"
+    android:textSize="50sp"/>
+
+
+
+</androidx.coordinatorlayout.widget.CoordinatorLayout>
```
Comment (tobireith) [inline app/src/main/res/layout/activity_plant.xml]: No newline ;)

File: app/src/main/res/navigation/nav_graph2.xml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+        android:id="@+id/SecondFragment"
+        android:name="de.othr.plantico.SecondFragment"
+        android:label="@string/second_fragment_label"
+        tools:layout="@layout/fragment_second" >
+
+        <action
+            android:id="@+id/action_SecondFragment_to_FirstFragment"
+            app:destination="@id/FirstFragment" />
+    </fragment>
+</navigation>
```
Comment (tobireith) [inline app/src/main/res/navigation/nav_graph2.xml]: No newline ;)

File: app/src/main/res/values/image_cactures_background.xml [hunk 13]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="image_cactures_background">#FFFFFF</color>
+</resources>
```
Comment (tobireith) [inline app/src/main/res/values/image_cactures_background.xml]: No newline ;)

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -13,4 +13,6 @@
     <string name="title_home">Home</string>
     <string name="title_dashboard">Dashboard</string>
     <string name="title_notifications">Notifications</string>
+    <string name="add_plant">Add a plant!</string>
+    <string name="title_activity_plant">PlantActivity</string>
 </resources>
```
Comment (tobireith) [inline app/src/main/res/values/strings.xml]: No newline ;)

**Review by tobireith — APPROVED** [review]
Mostly newlines, looks good :)

**Comment by tobireith** [comment]
@hoffmatteo I think we should also add a Test Class for each DAO. So in your case it would be a PlantDAOTest Class.

**Review by emuguy1 — APPROVED** [review]

**Pull request merged**